variables.tf

```diff
@@ -0,0 +1,290 @@
+
+variable "resource_group_tags" {
+  description = "Custom tags for the resource group"
+  type        = map(string)
+  default     = {}
+}
+
+variable "location" {
+  description = "Location of the resource group and other services in this module."
+  type        = string
+}
+
+#########################################
+# Variables related to Key Vault
+#########################################
+
+variable "enabled_for_deployment" {
+  description = "If Azure VM is permitted to retrieve secrets"
+  type        = bool
+  default     = false
+}
+
+variable "enabled_for_template_deployment" {
+  description = "If Azure RM is permitted to retrieve secrets"
+  type        = bool
+  default     = false
+}
+
+variable "soft_delete_retention_days" {
+  description = "Number of retention days for soft delete"
+  type        = number
+  default     = 7
+}
+
+variable "purge_protection_enabled" {
+  description = "If purge_protection is enabled"
+  type        = bool
+  default     = false
+}
+
+variable "sku_name" {
+  description = "SKU for the key vault - standard or premium"
+  type        = string
+  default     = "standard"
+}
+
+variable "custom_tags" {
+  description = "Custom tags for the Key vault"
+  type        = map(string)
+  default     = {}
+}
+
+variable "access_policies" {
+  description = "Additional Access policies for the vault except the current user which are added by default"
+  type = map(object({
+    object_id               = string
+    tenant_id               = string
+    key_permissions         = list(string)
+    certificate_permissions = list(string)
+    secret_permissions      = list(string)
+    storage_permissions     = list(string)
+  }))
+
+  default = {}
+}
+
+variable "enable_rbac_authorization" {
+  description = "Enable RBAC authorization for the key vault"
+  type        = bool
+  default     = false
+}
+
+variable "network_acls" {
+  description = "Network ACLs for the key vault"
+  type = object({
+    bypass                     = string
+    default_action             = string
+    ip_rules                   = optional(list(string))
+    virtual_network_subnet_ids = optional(list(string))
+  })
+
+  default = {
+    bypass                     = "AzureServices"
+    default_action             = "Allow"
+    ip_rules                   = []
+    virtual_network_subnet_ids = []
+  }
+}
+
+variable "public_network_access_enabled" {
+  description = " (Optional) Whether public network access is allowed for this Key Vault. Defaults to true."
+  type        = bool
+  default     = true
+}
+
+#########################################
+# Variables related to Resource Names
+#########################################
+variable "resource_names_map" {
+  description = "A map of key to resource_name that will be used by tf-launch-module_library-resource_name to generate resource names"
+  type = map(object({
+    name       = string
+    max_length = optional(number, 60)
+  }))
+
+  default = {}
```

Comment:
The defaults should be defined here instead of asking the consumers to add that in. As we as developers know exactly what resource types are being used in this module. If the user wants, they can override the defaults in the tfvars but the defaults are required.